sql/log_event_server.cc

```diff
@@ -5708,8 +5710,26 @@ int Rows_log_event::do_apply_event(rpl_group_info *rgi)
       if (!table->in_use)
         table->in_use= thd;
 
+      /*
+        Exit early, and let the Event-level exit logic take care of the cleanup
+        and rollback.
+      */
+      if (rgi->rli->mi->using_parallel() &&
+          rgi->parallel_entry->stop_abrupt(rgi->rli) &&
+          rgi->parallel_entry->rgi_is_safe_to_terminate(rgi))
+        break;
+
       error= do_exec_row(rgi);
```

Comment:
Why only in parallel replication? Changing STOP SLAVE to abort in the middle of a row event - if we really want that - would make sense for non-parallel as well as parallel?